### ISSUE TITLE
Encourage users to only download from official sources, and optionally verify downloads

### ISSUE THREAD

**Issue by kaie** [thundernest/thunderbird-website#490]
Somewhere in the download section, it would be good to explain to users how they can verify their downloads, in a discoverable way.

We should discuss what we should explain, which verification strategies we want to explain.

The intention is:

(1) create some general awareness that verifying downloads is a good idea
(the fact that such a verification offering can be found on the download page could be seen as a way to make users aware, and allow them to learn more, if they want to)

(2) Allow users a simple verification that could be done without downloading additional software. For example, if users find the SHA256 checksum on the download page, there could be a quick information how to use tools already available on the OS to verify (e.g. sha256sum on Linux and MacOS, and on Windows something like certutil -hashfile <file> SHA256)

(3) Potentially have a link that explains the more advanced checking. Which is, use GnuPG, and offer a link to the signature file.

**Comment by rtanglao** [comment]
* Do we need a SUMO KB article?

**Comment by kaie** [comment]
SUMO could be the place where the detailed explanations live.

==Download page==
- here is your download: link
- You may verify the correctness of your download using SHA256 hashsum ABCDEF000111... (dynamically embedded) or uising a GnuPG signature (dynamically set the correct link).
- Here are the instructions for verifying a download: link to SUMO
- remember to obtain thunderbird only from official locations, such as this thunderbird.net website or using a download mechanism that's integrated into your operating system.

**Comment by MelissaAutumn** [comment]
Some thoughts:

I don't believe we have a sha256 hash (or any hash) in product details which is a pre-req if we want it to appear on the site. https://github.com/mozilla-releng/product-details/tree/production

I could definitely see this being useful though. Some examples of how other sites handle this:

Ubuntu provides a pop-over card that displays instructions to verify the download:
<img width="744" alt="A thank you page with a pop-over card displaying how to verify your ubuntu server iso" src="https://github.com/thundernest/thunderbird-website/assets/97147377/2a2d2c96-5988-4f31-b457-2da60ec0d641">

openSUSE uses a dropdown with a link to the sha256 signature:
<img width="622" alt="A download button with a small arrow to the right, displaying a list which reads: metalink, pick mirror, and checksum." src="https://github.com/thundernest/thunderbird-website/assets/97147377/1790125a-193d-4e59-b990-93ee4e87ce71">


KDE Neon just has a link below the download button for the pgp signature:
<img width="470" alt="Below a download button appears the words PGP signature for verification" src="https://github.com/thundernest/thunderbird-website/assets/97147377/6cb4ae0f-cb9d-447b-8bc9-f66e1ff91b53">

**Comment by wsmwk** [comment]
request for checksums mentioned in https://chaos.social/@txt_file/114974783653100097

**Comment by MelissaAutumn** [comment]
Do you know (or maybe @coreycb) if we can pipe through the checksum to https://github.com/mozilla-releng/product-details/blob/production/public/1.0/thunderbird.json or if there's a magical bouncer url for a product checksum?

**Comment by coreycb** [comment]
I'm not sure off-hand. @dandarnell is this part of our taskgraph config?

**Comment by MelissaAutumn** [comment]
No rush if it isn't, just a nice to have eventually. 

If it is or once it's eventually added let us know and we can add it to the downloads page.

**Comment by coreycb** [comment]
I noticed that Firefox doesn't carry hashes in their product details: https://github.com/mozilla-releng/product-details/blob/production/public/1.0/firefox.json - would it make any sense to pull them from FTP or do they need to be in product details?

**Comment by Sancus** [comment]
> I noticed that Firefox doesn't carry hashes in their product details: https://github.com/mozilla-releng/product-details/blob/production/public/1.0/firefox.json - would it make any sense to pull them from FTP or do they need to be in product details?

To display the data on the website it needs to be cached somewhere, the FTP has it in one directory per build which means a lot of http requests for a single website build. Not good.

Product details seems like the most logical place to cache it... I don't think Firefox even shows their hashes on mozilla.org anywhere so they don't need this. But I guess we could cache it in this repository instead if adding it to product details is difficult for some reason.

**Comment by coreycb** [comment]
I think we can make the update [here](https://github.com/mozilla-releng/shipit/blob/3e05bb72a73280a447f1c3c97fe6f4a3dfaec6ca/api/src/shipit_api/admin/product_details.py#L1009)

I'll add a task to our notion backlog. Let me know if you have a time frame for needing this change and we can prioritize appropriately.

**Comment by janbrasna** [comment]
The TLDR is due to attribution bits changing the signature for a big part of Win&Mac demography the SHAsums are not exposed on firefox.com for downloads. They are basically only reasonably expected when folks go and download the installers directly from archive.m.o instead of bouncer in case of Fx…

Wondering if there's utility in storing the sums elsewhere. Folks will still need to calculate them somehow. Wouldn't it be better to show them a popup with the commands, and link to the appropriate e.g. https://archive.mozilla.org/pub/thunderbird/releases/142.0/ for given version with the sums? Or a trick command to curl-grep-awk-sed them to the correct checksum from those files, and compare to the one calculated for their download?
